packages/gestalt-datepicker/src/DateRange.js

```diff
@@ -0,0 +1,264 @@
+              {radioGroup}
+            </Box>
+          </div>
+        ) : null}
+        <Box>
+          <Flex alignItems="start" justifyContent="center" direction="column">
+            <div className={borderStyles.dateFieldSection}>
+              <Flex gap={3}>
+                <Box width={280}>
+                  <InternalDateField
+                    autoComplete="off"
+                    mobileEnterKeyHint="enter"
+                    id={`datefield-start-${componentId}`}
+                    localeData={localeData}
+                    onChange={({ value }) => {
+                      if (value?.getTime() || value === null) onStartDateChange({ value });
+                    }}
+                    onError={onStartDateError}
+                    onBlur={onStartDateBlur}
+                    onFocus={onStartDateFocus}
+                    value={startDateValue}
+                    minDate={minDate}
+                    maxDate={maxDate}
+                    errorMessage={startDateErrorMessage}
+                  />
+                </Box>
+                <Box dangerouslySetInlineStyle={{ __style: { marginTop: '15px' } }}>
+                  <Text>—</Text>
+                </Box>
+                <Box width={280}>
+                  <InternalDateField
+                    autoComplete="off"
+                    mobileEnterKeyHint="enter"
+                    id={`datefield-end-${componentId}`}
+                    localeData={localeData}
+                    onChange={({ value }) => {
+                      if (value?.getTime() || value === null) onEndDateChange({ value });
+                    }}
+                    value={endDateValue}
+                    minDate={startDateValue}
+                    onError={({ errorMessage, value }) => {
+                      onEndDateError({ errorMessage, value });
+                    }}
+                    onBlur={onEndDateBlur}
+                    onFocus={onEndDateFocus}
+                    maxDate={maxDate}
+                    errorMessage={endDateErrorMessage}
+                  />
+                </Box>
+              </Flex>
+            </div>
+            <Box width={629}>
+              <InternalDatePicker
+                localeData={localeData}
+                rangeStartDate={startDateValue}
+                rangeEndDate={endDateValue}
+                id={`datepicker-${componentId}`}
+                onChange={({ startDate, endDate }) => {
+                  onStartDateChange({ value: startDate });
+                  onEndDateChange({ value: endDate });
+                }}
+                minDate={minDate}
+                maxDate={maxDate}
+              />
+            </Box>
+            <Flex.Item alignSelf="end">
+              <Flex gap={2}>
+                <Box marginBottom={4} marginEnd={4}>
+                  <Button color="transparent" text={cancelText} onClick={() => onCancel()} />
+                </Box>
+                <Box marginBottom={4} marginEnd={4}>
```

Comment:
Question for Layla, I follow designs

Reply:
@laylaleeal

Reply:
The buttons should be a ButtonGroup, which has a set spacing between Buttons. We don't want to change that! and about the primary color, we had this discussion, but decided to go with a Secondary/Tertiary pair to avoid a case of having too many primary buttons in an interface (there's no wash applied when DateRange is open, so the page underneath is still visible).

Reply:
Cool!

Also, tiny feedback on the "Apply" language, which seems to be for a filter. Like, if I click apply I expect an action to happen immediately, e.g apply a filter, but in the case if it's used as part of larger form,  apply may not do immediate results. should we call it "done" like [Flights](https://www.google.com/travel/flights), and [Expedia](https://www.expedia.com/)

<img width="756" alt="image" src="https://github.com/pinterest/gestalt/assets/5509813/943fc21e-b8bd-4fa8-a3b1-cc68abeacc1d">

Reply:
@laylaleeal Apply or Done?

Reply:
sorry, just seeing this - I'm ok with Done! cc @hectoid any thoughts on it?

Reply:
@laylaleeal and @rlingineni I understand that travel sites use "Done", but the primary use case for us is Ads and Analytics. Our competitors use "Apply" or "Update". We currently use "Apply" elsewhere in the Pinterest Business interface. I think we should stick to that and can update if things change.
<img width="683" alt="image" src="https://github.com/pinterest/gestalt/assets/96082362/2d63090f-a1aa-4672-9637-112d1f4a51cd">

<img width="559" alt="image" src="https://github.com/pinterest/gestalt/assets/96082362/f6b5d4b7-57fc-4483-a410-e70ce5a848b0">

<img width="297" alt="image" src="https://github.com/pinterest/gestalt/assets/96082362/109eb572-8882-4eb3-bd37-75c29295de42">

Reply:
But, now I also think we should have a primary / secondary button treatment to match other dropdowns. If that is still a doable change @AlbertCarreras 
<img width="283" alt="image" src="https://github.com/pinterest/gestalt/assets/96082362/b77cd5ab-c5c2-4011-88c7-ed4b6bdcadb6">

Reply:
(also notice they're not following our Sentence case guidelines for buttons though)

Reply:
@laylaleeal @hectoid can we find consensus on the button treatment? https://github.com/pinterest/gestalt/pull/3001#discussion_r1265753701